Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -173,9 +173,75 @@ internal sealed class PythonLanguageClient : ILanguageClient, ILanguageClientCus
             customTarget.WatchedFilesRegistered += WatchedFilesRegistered;
             customTarget.WorkspaceFolderChangeRegistered += OnWorkspaceFolderWatched;
             customTarget.AnalysisComplete += OnAnalysisComplete;
+            customTarget.WorkspaceConfiguration += OnWorkspaceConfiguration;
             await StartAsync.InvokeAsync(this, EventArgs.Empty);
         }
 
+        private async Task OnWorkspaceConfiguration(object sender, WorkspaceConfiguration.ConfigurationArgs args) {
+            // This is where we send the settings for each client context.
+            // Pylance will send a workspace/configuration request for each workspace
+            // We return the language server settings as a response
+            List<object> result = new List<object>();
+
+            // Return the matching results
+            foreach (var item in args.requestParams.items) {
+                // Find the matching context for the item
+                var context = _clientContexts.Find(c => item.scopeUri != null && PathUtils.IsSamePath(c.RootPath, item.scopeUri.LocalPath));
+                if (context == null)
+                    continue;
+
+                if (context is PythonLanguageClientContextProject) {
+                    // Project interactions are main thread only.
+                    await JoinableTaskContext.Factory.SwitchToMainThreadAsync();
+                }
+
+                object settings = null;
+                if (item.section == "python") {
+                    Debug.Assert(_analysisOptions != null);
+
+                    var extraPaths = UserSettings.GetStringSetting(
+                        PythonConstants.ExtraPathsSetting, null, Site, PythonWorkspaceContextProvider.Workspace, out _)?.Split(';')
+                        ?? _analysisOptions.ExtraPaths;
+
+                    var stubPath = UserSettings.GetStringSetting(
+                        PythonConstants.StubPathSetting, null, Site, PythonWorkspaceContextProvider.Workspace, out _)
+                        ?? _analysisOptions.StubPath;
+
+                    var typeCheckingMode = UserSettings.GetStringSetting(
+                        PythonConstants.TypeCheckingModeSetting, null, Site, PythonWorkspaceContextProvider.Workspace, out _)
+                        ?? _analysisOptions.TypeCheckingMode;
+
+                    var ver3 = new Version(3, 0);
+                    if (context.InterpreterConfiguration.Version < ver3) {
+                        MessageBox.ShowWarningMessage(Site, Strings.WarningPython2NotSupported);
+                    }
+
+                    settings = new LanguageServerSettings.PythonSettings {
+                        pythonPath = context.InterpreterConfiguration.InterpreterPath,
```

Comment:
This one right here? Or did you mean something else?

Reply:
Did you mean this one?

```_envPYTHONPATH```

Reply:
Yes I think I can add that one too. I'll do _envPYTHONPATH

Reply:
Sorry it doesn't have that information. It just has the name of the 'PYTHONPATH' environment variable but not the actual values.

Reply:
Is it necessary for pylance to work correctly? I think I might be able to run the environment to extract it, but that's a lot more code.

Reply:
No, it's not necessary to run. If VS doesn't provide a way to configure this, then there's no reason to try and fetch it. VSC does, so I piped that down.